### ISSUE TITLE
Macos bindings

### ISSUE THREAD

**Issue by MrManiacc** [SpaiR/imgui-java#2]
Hi I'm trying to create a cross platform editor could you create mac bindings please?

**Comment by hwhea** [comment]
@SpaiR do mac bindings exist? If not can you provide some basic pointers on how to create them? Happy to contribute.

**Comment by SpaiR** [comment]
@hwhea Basicaly, to create bindings for mac you only have to compile native binaries. To do that locally you need to modify this [file](https://github.com/SpaiR/imgui-java/blob/master/buildSrc/src/main/groovy/imgui/generate/GenerateLibs.groovy) in the next way:
```
def mac = BuildTarget.newDefaultTarget(BuildTarget.TargetOs.MacOsX, is64bit)
new AntScriptGenerator().generate(buildConfig, mac)
BuildExecutor.executeAnt(jniDir + '/build-macosx32.xml', '-v', '-Dhas-compiler=true', '-Drelease=true', 'clean', 'postcompile')
```
If ant and mingw-w64 is installed, then run `./gradlew :imgui-binding:generateLibs` and you'll get your binaries.

The blocker right now is that it's impossible to do that from any system other than Mac. Or at least I didn't managed to find how to do that. And since I'm not the mac user, the propper infrastacture to build binaries remotly need to be configured (like github actions for example).

**Comment by hwhea** [comment]
@SpaiR thanks for the guide. After much trial and error, I've managed to generate the natives. I'm pretty new to gradle, so could you provide advice on how to include them in my gradle.build please?

**Comment by SpaiR** [comment]
@hwhea You don't need to add compiled binaries in build.gradle. Everything you need is to say your java application the path where to find them. That could be done with `java.library.path` property. You provide it as a JVM argument with the path to a folder where you've placed your libs. 

Let's say you have a fat jar with all dependencies included called `my-app.jar`. The proper command to start it will be: `java -Djava.library.path=/path/to/folder/with/bins -jar my-app.jar`.

You may also create a gradle task to do it locally, see `startExample` task https://github.com/SpaiR/imgui-java/blob/master/imgui-lwjgl3/build.gradle#L35-L39 (you would like to remove `classpath` line, yeah).

**Comment by hwhea** [comment]
@SpaiR Thanks for this. I actually had that done already and was getting an error, but it seems my error is related to GLSL versioning and Mac's drivers. I get the error:

`
2020-02-24 14:19 > GLVER:4.10
Exception in thread "main" java.lang.IllegalStateException: ERROR: 0:1: '' :  version '130' is not supported
ERROR: 0:2: '' :  #version required and missing.

	at imgui.gl3.ImGuiImplGl3.loadAndCompileShader(ImGuiImplGl3.java:322)
	at imgui.gl3.ImGuiImplGl3.prepareShader(ImGuiImplGl3.java:182)
	at imgui.gl3.ImGuiImplGl3.init(ImGuiImplGl3.java:68)
	at com.redwireinteractive.firefly.test.ImGuiGlfwExample.initImGui(ImGuiGlfwExample.java:257)
	at com.redwireinteractive.firefly.test.ImGuiGlfwExample.run(ImGuiGlfwExample.java:64)
	at com.redwireinteractive.firefly.test.ImGuiGlfwExample.main(ImGuiGlfwExample.java:461)
ERROR: 0:2: '' :  #version required and missing.


> Task :ImGuiGlfwExample.main() FAILED

`

Any idea how I can force IMGUI to use a different version?

**Comment by SpaiR** [comment]
Duh.. MacOS doesn't support OpenGL 3.0 profile (shaders in lwjgl renderer are hardcoded to use it). For now you can copy and paste lwjgl3 renderer [class](https://github.com/SpaiR/imgui-java/blob/master/imgui-lwjgl3/src/main/java/imgui/gl3/ImGuiImplGl3.java) with shader files it uses (could be found [here](https://github.com/SpaiR/imgui-java/tree/master/imgui-lwjgl3/src/main/resources)). Then  replace `#version 130` with `#version 150` in both .frag and .vert files. And instead of mine implementation use the copy-pasted one. Don't forget to remove gradle dependecies to lwjgl3 renderer from my library, since in your case it would be unneeded.

In the near future I will fix this moment.

**Comment by hwhea** [comment]
Success! Finally --- well, sort of. I've finally got a window, but no IMGUI UI is showing - I think it's a shader problem. I've attached shaders below:

Vert:
```
#version 150

uniform mat4 ProjMtx;
in vec2 Position;
in vec2 UV;
in vec4 Color;
out vec2 Frag_UV;
out vec4 Frag_Color;

void main()
{
    Frag_UV = UV;
    Frag_Color = Color;
    gl_Position = ProjMtx * vec4(Position.xy, 0, 1);
}
```

Frag:
```
#version 150

#ifdef GL_ES
precision mediump float;
#endif

uniform sampler2D Texture;
in vec2 Frag_UV;
in vec4 Frag_Color;

out vec4 fragColor;
void main()
{
    fragColor = Frag_Color * texture(Texture, Frag_UV.st);
}

```

I am very new to shader programming so I'm struggling to find the problem. In the frag shader I changed texture2d to texture as texture2d was throwing an error (deprecated).

**Comment by hwhea** [comment]
Extra info: When I hover in certain places, I can see different cursors. That would suggest that the text elements/ buttons etc are there, but they are not rendering.

**Comment by MrManiacc** [comment]
I'm going to look into this later today, because I have a mac, and I use it half the time for dev of my engine. I'm starting to implement this imgui port into my engine today, so If I get a working revision of the imgui on mac I'll send a pull request.

**Comment by MrManiacc** [comment]
> I'm going to look into this later today, because I have a mac, and I use it half the time for dev of my engine. I'm starting to implement this imgui port into my engine today, so If I get a working revision of the imgui on mac I'll send a pull request.

Update, I've gotten it working other than input which is weird, i'm going to send a pull request now.

**Comment by SpaiR** [comment]
MacOS binding is available since [v1.76-0.11](https://github.com/spair/imgui-java/tree/v1.76-0.11)
